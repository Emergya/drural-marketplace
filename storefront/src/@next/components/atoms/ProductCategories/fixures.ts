import { ProductDetails_categories_edges_node } from "@drural/sdk/lib/fragments/gqlTypes/ProductDetails";

export const PRODUCT_CATEGORIES: ProductDetails_categories_edges_node[] = [
  {
    __typename: "Category",
    id: "Q2F0ZWdvcnk6Nw==",
    name: "Art & books",
    slug: "art-and-books",
    products: null,
  },
  {
    __typename: "Category",
    id: "Q2F0ZWdvcnk6Nw==",
    name: "Art & books",
    slug: "art-and-books",
    products: null,
  },
  {
    __typename: "Category",
    id: "Q2F0ZWdvcnk6Nw==",
    name: "Art & books",
    slug: "art-and-books",
    products: null,
  },
];
