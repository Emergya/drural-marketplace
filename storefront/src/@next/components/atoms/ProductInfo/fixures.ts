import { IProps } from "./types";

export const DEFAULT_PROPS: IProps = {
  title: "Description",
  data: `{"time": 1635490805627, "blocks": [{"data": {"text": "This is the product description"}, "type": "paragraph"}], "version": "2.20.0"}`,
};
