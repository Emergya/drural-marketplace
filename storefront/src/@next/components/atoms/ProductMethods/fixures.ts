import { IProps, MethodIdentifierEnum } from "./types";

export const DEFAULT_PROPS: IProps = {
  title: "Payment methods",
  methods: [
    {
      __typename: "PaymentMethod",
      id: "1",
      name: "Bank transfer",
      identifier: MethodIdentifierEnum.BANK_TRANSFER,
      isActive: true,
    },
    {
      __typename: "PaymentMethod",
      id: "1",
      name: "Card",
      identifier: MethodIdentifierEnum.CARD,
      isActive: true,
    },
    {
      __typename: "PaymentMethod",
      id: "1",
      name: "Pay at store",
      identifier: MethodIdentifierEnum.PAY_AT_STORE,
      isActive: true,
    },
  ],
};
