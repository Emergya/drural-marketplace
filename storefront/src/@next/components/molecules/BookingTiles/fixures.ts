export const resourceCalendarAvailability = [
  {
    date: "2021-12-27",
    available: true,
  },
  {
    date: "2021-12-28",
    available: false,
  },
  {
    date: "2021-12-29",
    available: true,
  },
  {
    date: "2021-12-30",
    available: false,
  },
  {
    date: "2021-12-31",
    available: true,
  },
  {
    date: "2022-01-01",
    available: false,
  },
  {
    date: "2022-01-02",
    available: false,
  },
  {
    date: "2022-01-03",
    available: true,
  },
  {
    date: "2022-01-04",
    available: false,
  },
  {
    date: "2022-01-05",
    available: true,
  },
  {
    date: "2022-01-06",
    available: false,
  },
  {
    date: "2022-01-07",
    available: true,
  },
  {
    date: "2022-01-08",
    available: false,
  },
  {
    date: "2022-01-09",
    available: false,
  },
  {
    date: "2022-01-10",
    available: false,
  },
  {
    date: "2022-01-11",
    available: false,
  },
  {
    date: "2022-01-12",
    available: false,
  },
  {
    date: "2022-01-13",
    available: false,
  },
  {
    date: "2022-01-14",
    available: false,
  },
  {
    date: "2022-01-15",
    available: false,
  },
  {
    date: "2022-01-16",
    available: false,
  },
  {
    date: "2022-01-17",
    available: true,
  },
  {
    date: "2022-01-18",
    available: false,
  },
  {
    date: "2022-01-19",
    available: true,
  },
  {
    date: "2022-01-20",
    available: false,
  },
  {
    date: "2022-01-21",
    available: true,
  },
  {
    date: "2022-01-22",
    available: false,
  },
  {
    date: "2022-01-23",
    available: false,
  },
  {
    date: "2022-01-24",
    available: true,
  },
  {
    date: "2022-01-25",
    available: false,
  },
  {
    date: "2022-01-26",
    available: true,
  },
  {
    date: "2022-01-27",
    available: false,
  },
  {
    date: "2022-01-28",
    available: true,
  },
  {
    date: "2022-01-29",
    available: false,
  },
  {
    date: "2022-01-30",
    available: false,
  },
  {
    date: "2022-01-31",
    available: true,
  },
  {
    date: "2022-02-01",
    available: false,
  },
  {
    date: "2022-02-02",
    available: true,
  },
  {
    date: "2022-02-03",
    available: false,
  },
  {
    date: "2022-02-04",
    available: true,
  },
];

export const resourceAvailabilityByDate = [
  {
    startTime: "01:45:00",
    endTime: "02:15:00",
  },
  {
    startTime: "02:15:00",
    endTime: "02:45:00",
  },
  {
    startTime: "02:45:00",
    endTime: "03:15:00",
  },
  {
    startTime: "07:45:00",
    endTime: "08:15:00",
  },
  {
    startTime: "08:15:00",
    endTime: "08:45:00",
  },
];
