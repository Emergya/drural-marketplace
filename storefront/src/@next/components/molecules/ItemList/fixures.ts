export const collections = [
  {
    __typename: "Collection",
    id: "1",
    name: "Collection 1",
    slug: "collection-1",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "2",
    name: "Collection 2",
    slug: "collection-2",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "3",
    name: "Collection 3",
    slug: "collection-3",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "4",
    name: "Collection 4",
    slug: "collection-4",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "5",
    name: "Collection 5",
    slug: "collection-5",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "6",
    name: "Collection 6",
    slug: "collection-6",
    backgroundImage: null,
  },
];
