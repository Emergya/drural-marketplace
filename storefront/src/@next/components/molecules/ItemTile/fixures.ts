export const collection = {
  __typename: "Collection",
  id: "1",
  name: "Collection 1",
  slug: "collection-1",
  backgroundImage: null,
};

export const category = {
  __typename: "Category",
  id: "1a",
  name: "Category 1",
  slug: "category-1",
  iconId: null,
};
