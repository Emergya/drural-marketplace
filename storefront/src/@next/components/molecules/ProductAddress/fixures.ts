import { ProductAddressCountry } from "@drural/sdk";
import { ProductDetails_address } from "@drural/sdk/lib/fragments/gqlTypes/ProductDetails";

export const PRODUCT_ADDRESS: ProductDetails_address = {
  __typename: "ProductAddressType",
  street: "Pascal",
  postalCode: "28034",
  locality: "Madrid",
  region: "Madrid",
  country: ProductAddressCountry.ES,
  latitude: 37.37301,
  longitude: -5.74951,
};
