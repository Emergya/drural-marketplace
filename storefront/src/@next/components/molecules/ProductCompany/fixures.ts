import { ProductDetails_company } from "@drural/sdk/lib/fragments/gqlTypes/ProductDetails";

export const PRODUCT_COMPANY: ProductDetails_company = {
  __typename: "CompanyType",
  id: "Q29tcGFueVR5cGU6MQ==",
  imageUrl: "/media/companies/dRural_-_Marketp_11.png",
  name: "Comany 01",
  publicName: "Comany 01",
  rating: 2,
  products: {
    __typename: "ProductCountableConnection",
    totalCount: 3,
  },
};
