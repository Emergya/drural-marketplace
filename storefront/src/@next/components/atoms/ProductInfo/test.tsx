import "jest-styled-components";

import { mount, ReactWrapper } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { dRuralTheme } from "@styles";

import { ProductInfo } from ".";
import { DEFAULT_PROPS } from "./fixures";

describe("<ProductInfo />", () => {
  // 1. Variables
  let wrapper: ReactWrapper;

  // 2. Wrapper
  beforeEach(() => {
    wrapper = mount(<ProductInfo {...DEFAULT_PROPS} />);
  });

  // 3. Tests
  it("exists", () => {
    expect(wrapper.exists()).toEqual(true);
  });

  it("contains provided data", () => {
    const keys = Object.keys(DEFAULT_PROPS);

    keys.map(value => {
      if (value !== "data") {
        expect(wrapper.text()).toContain((DEFAULT_PROPS as any)[value]);
      }
    });
    expect(wrapper.text()).toContain("This is the product description");
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={dRuralTheme}>
          <ProductInfo {...DEFAULT_PROPS} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
