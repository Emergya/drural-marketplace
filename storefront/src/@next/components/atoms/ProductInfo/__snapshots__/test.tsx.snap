// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInfo /> renders correctly 1`] = `
.c2 .ce-block__content {
  max-width: 100%;
}

.c0:not(:last-child) {
  padding-bottom: 3rem;
}

.c0:last-child {
  padding-bottom: 1rem;
}

.c1 {
  margin-bottom: 14px;
}

@media (max-width:540px) {
  .c0:not(:last-child) {
    padding-bottom: 2rem;
  }
}

@media (max-width:540px) {
  .c0:last-child {
    padding-bottom: 10px;
  }
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    Description
  </h3>
  <div
    className="c2"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>This is the product description</p>",
      }
    }
  />
</div>
`;
